src/pcre2_compile.c

```diff
@@ -5736,8 +5739,10 @@ for (;; pptr++)
     /* ===================================================================*/
     /* Empty character classes are allowed if PCRE2_ALLOW_EMPTY_CLASS is set.
     Otherwise, an initial ']' is taken as a data character. When empty classes
-    are allowed, [] must always fail, so generate OP_FAIL, whereas [^] must
-    match any character, so generate OP_ALLANY. */
+    are allowed, [] must generate an empty class - we have no dedicated opcode
+    to optimise the representation, but it's a rare (and useless?) case. The
```

Comment:
I would say `(*F)` is a much better option to fail an alternative.

Reply:
I think you updated this code recently? I just noticed that the comment seemed not to match.